cmd/duffle/claims_show.go

```diff
@@ -15,42 +17,66 @@ Display the content of a claim.
 This dumps the entire content of a claim as a JSON object.
 `
 
+type claimsShowCmd struct {
+	Name       string
+	OnlyBundle bool
+	Output     string
+	Storage    claim.Store
+}
+
 func newClaimsShowCmd(w io.Writer) *cobra.Command {
-	var onlyBundle bool
+	var cmdData claimsShowCmd
 	cmd := &cobra.Command{
 		Use:     "show NAME",
 		Short:   "show a claim",
 		Long:    claimsShowDesc,
 		Aliases: []string{"get"},
 		Args:    cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			name := args[0]
-			storage := claimStorage()
-			return displayClaim(name, w, storage, onlyBundle)
+			cmdData.Name = args[0]
+			cmdData.Storage = claimStorage()
+			return cmdData.runClaimShow(w)
 		},
 	}
 
-	cmd.Flags().BoolVarP(&onlyBundle, "bundle", "b", false, "only show the bundle from the claim")
+	cmd.Flags().BoolVarP(&cmdData.OnlyBundle, "bundle", "b", false, "only show the bundle from the claim")
```

Comment:
Why not make this show the bundle from the claim, but not necessarily only the bundle? Then users would mix `-o` and `-b` if they wanted to?

Reply:
A reason not to enable `-o` and `-b` to be used together is that it is not obvious what such a command should do.

Some options:
* If you think of each of these as a filter on what output should be produced, then specifying both gives you no output (because they filter for different parts of the claim, there is not intersection).
* Since of `-o` and `-b` has a specific output, using both gives you both (in some order).
* Since `-b` says you want part of the bundle and `-o` selects an output listed in the bundle, both gives the specification of the output from the bundle in the claim (rather than the contents of the file).

We tend to think of the CLI behavior through the first mental model (filtering), so the use-case of combining the flags is very limited. However, if users did want to mix `-o` and `-b`, we should decide what that behavior should be and how to easily convey that to the user.

Reply:
I was also wondering if `--output` could be a string slice, and then print all the selected outputs ? 🤔

Reply:
I'm fine fine with either a string slice or the way it's implemented in the PR; both implementations fit our use case. I leave it to the repo owners to decide what the cli user experience should be.